pkg/sql/backfill/backfill.go

```diff
@@ -360,6 +359,15 @@ type IndexBackfiller struct {
 	types   []*types.T
 	rowVals tree.Datums
 	evalCtx *tree.EvalContext
+	cols    []sqlbase.ColumnDescriptor
+
+	// predicates is a map of indexes to partial index predicate expressions. It
+	// includes entries for partial indexes only.
+	predicates map[sqlbase.IndexID]tree.TypedExpr
+	// indexesToEncode is a list of indexes to encode entries for a given row.
+	// It is a field of IndexBackfiller to avoid allocating a slice for each row
+	// backfilled.
+	indexesToEncode []sqlbase.IndexDescriptor
```

Comment:
this maybe should be a slice of pointers -- copying descriptors here seems somewhat expensive (since we're going to do it per row)

Reply:
It has to be a `[]sqlbase.IndexDescriptor` unless we change the type signature of `sqlbase.EncodeSecondaryIndexes`. I can go ahead and do that, but it'll be a pretty big change.

Reply:
Let's measure it and see (see my comment below). If it does show up as expensive, then I think adding an another function that takes all of the indexes, and then a slice of index ordinals to write would be clean and not affect the API too much.

Reply:
I tested **(A)** regular index backfill on 100k rows on master vs. **(B)** a partial index backfill on 100k rows where every row is in the partial index.

#### A

<img width="986" alt="master" src="https://user-images.githubusercontent.com/1128750/89452352-5d204300-d712-11ea-8e51-17b45b374a92.png">

#### B

<img width="984" alt="partial_index_backfill" src="https://user-images.githubusercontent.com/1128750/89452366-61e4f700-d712-11ea-9584-c3459b1a4d61.png">

Reply:
I've created an issue to track reducing these allocations in the future: https://github.com/cockroachdb/cockroach/issues/52425